### PR TITLE
feat: hold structurizr

### PR DESCRIPTION
**Pull request by edubacco**
It's a long time we do not update them.

Even the last ArCo group used a different tool to document their vision of the desired architecture (see https://github.com/casavo/community-of-practice/tree/master/software-architecture/diagrams)

If we don't use it, let's hold it.

### CODE REVIEW

**Comment by pikachuCa** [comment]
we are using **https://diagrams.mingrammer.com/docs/getting-started/installation** on our side.

Maybe we could keep just one ?

**Comment by xpepper** [comment]
> It's a long time we do not update them.
> 
> Even the last ArCo group used a different tool to document their vision of the desired architecture (see https://github.com/casavo/community-of-practice/tree/master/software-architecture/diagrams)
> 
> If we don't use it, let's hold it.

I agree to put in on "Hold".

**Comment by xpepper** [comment]
> we are using **https://diagrams.mingrammer.com/docs/getting-started/installation** on our side.
> 
> Maybe we could keep just one ?

Is an alternative to Structuriz? What are its main benefits?

**Comment by pferretti** [comment]
@edubacco if [this](https://github.com/casavo/c4-model) is the repo for our architectural diagrams, I agree that we are not using Structurizr (the last commit on master is more or less 7 month ago). So for me it's ok to put this **on hold**.

I think Excalidraw is a very good tool, very straightforward for drawing diagrams. However, I think it can only be a temporary substitute, because although the Excalidraw files are plain-text and therefore versionable, the sources are not suitable to be edited by a human.

We can evaluate the tool we use in Paris Squad, proposed by @pikachuCa.

In my opinion, a tool for drawing architectural diagrams must:
- be immediate in use, at least for a software engineer
- offer plain-text output, thus easily versionable
- offer output that is easily editable by a human being
- offer output that is automatically generated by the sources, that is comprehensible (and possibly pleasant to read)

Basically, friction in the use of the tool must be minimal, so that teams are incentivised to keep architectural diagrams up-to-date. In addition, the fact that the diagram is generated from the sources minimises the cognitive load for those who design/update the diagram, as they have to deal with the content, not the form.

What do you think @edubacco @pikachuCa @xpepper @dottiOliviero @YoannDelpierre @moebiusmania @enricobenetti ?

**Review by pferretti — APPROVED** [review]

**Comment by dottiOliviero** [comment]
if i try to open one i get 
image
I think it works locally. We can put it on Hold and specify that we need to find a substitute, with the characteristics that @pferretti listed above.

File: radar/casavo-tech-radar.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 [
   {
     "name": "structurizr",
-    "ring": "adopt",
+    "ring": "hold",
     "quadrant": "tools",
-    "isNew": "FALSE",
-    "description": "Diagrams for architecture ..."
+    "isNew": "TRUE",
+    "description": "Diagrams for architecture. Not used for a long time, maintaining is complex. Hold"
```
Comment (dottiOliviero) [inline radar/casavo-tech-radar.json:7]:
```suggestion
    "description": "Diagrams for architecture. Not used for a long time, maintaining is complex. We need to find a better alternative, that is immediate to use, easily versioned, offers an output that is easily editable by a human being and offers an output that is automatically generated by the sources. Hold"
```

**Review by xpepper — APPROVED** [review]

**Review by xpepper — APPROVED** [review]
I agree to put this on HOLD.

**Comment by xpepper** [comment]
> @edubacco if [this](https://github.com/casavo/c4-model) is the repo for our architectural diagrams, I agree that we are not using Structurizr (the last commit on master is more or less 7 month ago). So for me it's ok to put this **on hold**.
> 
> I think Excalidraw is a very good tool, very straightforward for drawing diagrams. However, I think it can only be a temporary substitute, because although the Excalidraw files are plain-text and therefore versionable, the sources are not suitable to be edited by a human.
> 
> We can evaluate the tool we use in Paris Squad, proposed by @pikachuCa.
> 
> In my opinion, a tool for drawing architectural diagrams must:
> 
> * be immediate in use, at least for a software engineer
> * offer plain-text output, thus easily versionable
> * offer output that is easily editable by a human being
> * offer output that is automatically generated by the sources, that is comprehensible (and possibly pleasant to read)
> 
> Basically, friction in the use of the tool must be minimal, so that teams are incentivised to keep architectural diagrams up-to-date. In addition, the fact that the diagram is generated from the sources minimises the cognitive load for those who design/update the diagram, as they have to deal with the content, not the form.
> 
> What do you think @edubacco @pikachuCa @xpepper @dottiOliviero @YoannDelpierre @moebiusmania @enricobenetti ?

About the alternatives to Structurizr, I would also propose **Mermaid** (https://mermaid.js.org), which supports C4 (https://mermaid.js.org/syntax/c4.html#c4-diagrams) and keeps being updated with new formats...

Having said this, if we already tested https://diagrams.mingrammer.com and we're having good feedbacks, it's ok to use it... we could put "Diagrams" in "Trial" maybe @edubacco @pikachuCa ?

**Review by pikachuCa — APPROVED** [review]

**Comment by edubacco** [comment]
> @edubacco if [this](https://github.com/casavo/c4-model) is the repo for our architectural diagrams, I agree that we are not using Structurizr (the last commit on master is more or less 7 month ago). So for me it's ok to put this **on hold**.
> 
> I think Excalidraw is a very good tool, very straightforward for drawing diagrams. However, I think it can only be a temporary substitute, because although the Excalidraw files are plain-text and therefore versionable, the sources are not suitable to be edited by a human.
> 
> We can evaluate the tool we use in Paris Squad, proposed by @pikachuCa.
> 
> In my opinion, a tool for drawing architectural diagrams must:
> 
> * be immediate in use, at least for a software engineer
> * offer plain-text output, thus easily versionable
> * offer output that is easily editable by a human being
> * offer output that is automatically generated by the sources, that is comprehensible (and possibly pleasant to read)
> 
> Basically, friction in the use of the tool must be minimal, so that teams are incentivised to keep architectural diagrams up-to-date. In addition, the fact that the diagram is generated from the sources minimises the cognitive load for those who design/update the diagram, as they have to deal with the content, not the form.
> 
> What do you think @edubacco @pikachuCa @xpepper @dottiOliviero @YoannDelpierre @moebiusmania @enricobenetti ?

I totally agree.

I do not have experience with mermaid, nor with mingrammer. But if we have minegrammer already up and running, we should add it to the radar and try it also on casavo side.
